### PR TITLE
raidboss: more e12s advanced triggers

### PR DESCRIPTION
**Pull request by quisquous**

### CODE REVIEW

**Comment by quisquous** [comment]
@JLGarber did this all look good to you? I added a couple more triggers while I was thinking about it too.  Gotta get in those critical week 16 triggers.  <_<

**Review by JLGarber — APPROVED** [review]

File: ui/raidboss/data/05-shb/raid/e12s.js [hunk 1]
```diff
@@ -1391,6 +1408,133 @@ export default {
...
+          en: 'Knockback Into Spread',
+        },
+      },
+    },
+    {
+      id: 'E12S Dark Water Stacks',
+      netRegex: NetRegexes.gainsEffect({ effectId: '99D' }),
+      // During Advanced Relativity, there is a very short Dark Water III stack (12s)
+      // that applies when people position themselves for the initial Return placement.
+      // Most strategies auto-handle this, and so this feels like noise.  MOREVER,
```
[the remark this answers is not in the document]
Comment (quisquous) [reply ui/raidboss/data/05-shb/raid/e12s.js]: Whoops, fixed.

File: ui/raidboss/data/05-shb/raid/e12s.js [hunk 2]
```diff
@@ -1391,6 +1408,133 @@ export default {
...
+      // using suppress here without this conditional will pick one of the short/long
+      // Dark Water III buffs and suppress the other, so this is a load-bearing conditional.
+      condition: (data, matches) => data.phase && parseFloat(matches.duration) > 13,
```
[the remark this answers is not in the document]
Comment (quisquous) [reply ui/raidboss/data/05-shb/raid/e12s.js:1469]: I left an extra comment there.  I think it makes more sense when compared with the `!data.phase` conditional on the other water stacks trigger.

File: ui/raidboss/data/05-shb/raid/e12s.js [hunk 3]
```diff
@@ -1391,6 +1408,133 @@ export default {
...
+      // In advanced, Aero comes in ~23 and ~31s flavors
+      condition: (data, matches) => data.phase === 'advanced' && parseFloat(matches.duration) > 28,
+      infoText: (data, matches, output) => {
+        data.doubleAero = data.doubleAero || [];
+        data.doubleAero.push(data.ShortName(matches.target));
+
+        if (data.doubleAero.length !== 2)
+          return;
+
+        return output.text({ name1: data.doubleAero[0], name2: data.doubleAero[1] });
```
[the remark this answers is not in the document]
Comment (quisquous) [reply ui/raidboss/data/05-shb/raid/e12s.js]: Done!

**Pull request merged**